core/src/main/scala/kanaloa/reactive/dispatcher/queue/Worker.scala

```diff
@@ -39,22 +37,22 @@ trait Worker extends Actor with ActorLogging with MessageScheduler {
     }
   }
 
-  def starting(work: Option[(Work, ActorRef)], askWorkImmediately: Boolean): Receive = whileWaiting(Starting) orElse {
+  def waitingForRoutee(work: Option[(Work, ActorRef)]): Receive = whileWaiting(Starting) orElse {
```

Comment:
So, these changes I don't think actually introduced any new issues with this state (unless I'm misunderstanding what you are saying), this "flapping" state already existed, and prior to the last PR, where if the Worker was in this state and received work, while it already had Work, it would overwrite it.